packages/http-client/src/index.ts

```diff
@@ -632,12 +632,69 @@ export class HttpClient {
     additionalHeaders: http.OutgoingHttpHeaders,
     header: string,
     _default: string
-  ): string | number | string[] {
+  ): string | string[] {
```

Comment:
It's not obvious without deeply studying this code what the gist of the change is about here. Can you either add some PR comments explaining this diff or some comments in the source code (or both)?

Reply:
Does this change behavior in any meaningful way? It looks like we might be doing `join(', ')` in some places we previously didn't. Is that something we need to document in the release notes?